### PR TITLE
Milestone 1

### PR DESCRIPTION
**Pull request by andgarzonmal**
# Milestone 1 📚

- [x] Create class Person with the following:

- Instance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- [x]  Create class Student with the following:

- Inherits from Person.
- Constructor extends parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".

- [x] Create class Teacher with the following:

- Inherits from Person.
- Constructor extends parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.

### CODE REVIEW

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @andgarzonmal , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project without a descriptive readme file

- [x] Please use this Microverse [Readme Template](https://docs.google.com/document/d/12UJLx0iJYSyENzM6-GICVsvmRxCZ3YqaWmqvi2xPaXo/edit#) update your readme file before submitting for next review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: classes/person.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Person
+  attr_accessor :id, :name, :age
```
Comment (ShahierNashaat) [inline classes/person.rb]: - [x] Kindly notice that the id should only be able to be retrieved so only the Getter is defined.
```ruby
attr_reader :id
```

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @andgarzonmal, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/person.rb (hunk 1 above)
Comment (andgarzonmal) [reply classes/person.rb]: Hi @ShahierNashaat  thanks for your comment i will fix it

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @andgarzonmal,

✔️ Good documentation

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**